### PR TITLE
always add trailing slash to base path

### LINKED ISSUE #9341: SvelteKit 1.9 and 1.10 using "base" config parameter and static adapter creates wrong imports (falsely prependended by ".").
### Describe the bug

Using the static adapter (sveltejs/adapter-static), Sveltekit builds wrong links in the created HTML files. 
The "base" parameter ist set to "build" so that the application is found in a sub-path "/build".
See "base" parameter under https://kit.svelte.dev/docs/configuration#paths

In sveltejs/kit Version 1.8.8 a typical link in a created index.html looks like that:

```html
<link rel="modulepreload" href="/build/_app/immutable/entry/start.84521e90.js">
```

In sveltejs/kit Version 1.10.0 and 1.9.0 it looks like that:
```html
<link rel="modulepreload" href="./build/_app/immutable/entry/start.84521e90.js">
```

A dot (".") is prepended the created href, leading to a wrong path:
when accessing "http://localhost/build/index.html", the dot leads to a path reference to "http://localhost/build/build/_app/immutable/entry/start.84521e90.js"

It seems this behavior makes the "base" parameter obsolete, as the sveltekit app is always accessible in any sub directory without any further need of configuration, because no absolute paths are used anymore.

That new behaviour is great!
But only if the base parameter is not set, otherwise the site is broken.

Still, in the documentation 
https://kit.svelte.dev/docs/configuration#paths
there is written to use the "base" parameter in the case of non-root-paths.

If I understand everything right, the solution can simply be to update the documentation, that no base parameter is needed anymore (remove it from documentation).
Also, an existing base parameter must be ignored and marked deprecated, or lead to an error with the message, that this parameter is not available/allowed anymore.

### Reproduction

I am using sveltejs/kit 1.9.0 or 1.10.0 and sveltejs/adapter-static 2.0.1

The vite config is:

```js
import { sveltekit } from '@sveltejs/kit/vite'

const config = {
	plugins: [sveltekit(),],
	server: {
		port: 3000,
		fs: {
			// Allow serving files from one level up to the project root
			allow: ['..']
		},
		base: "/build"
	},
	preview: {
		port: 3000
	},
	test: {
		globals: true
	}
}
export default config
```

The Svelte configuration is:
```js
import adapter_static from '@sveltejs/adapter-static'
import preprocess from 'svelte-preprocess'

const config = {
	preprocess: preprocess(),
	onwarn(warning, defaultHandler) {
		defaultHandler(warning)
	},
	kit: {
		adapter: adapter_static(),
		paths: {
			base: "/build"
		}
	}
}
export default config
```

### Logs

_No response_

### System Info

```Shell
System:
    OS: Linux 6.1 Ubuntu 22.10 22.10 (Kinetic Kudu)
    CPU: (16) x64 AMD Ryzen 7 6800U with Radeon Graphics
    Memory: 2.17 GB / 13.33 GB
    Container: Yes
    Shell: 5.2.2 - /bin/bash
  Binaries:
    Node: 19.7.0 - /usr/bin/node
    npm: 9.5.0 - /usr/bin/npm
  Browsers:
    Chrome: 110.0.5481.177
    Firefox: 110.0.1
  npmPackages:
    @sveltejs/adapter-auto: ^2.0 => 2.0.0 
    @sveltejs/adapter-node: ^1.2 => 1.2.1 
    @sveltejs/adapter-static: ^2.0 => 2.0.1 
    @sveltejs/kit: 1.9.0 => 1.9.0 
    svelte: ^3.55 => 3.55.1
```

### Severity

serious, but I can work around it

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by Rich-Harris**
fixes #9341.

In #9220, we changed the logic for resolving asset paths to accommodate the `relative: true` option, and included [special handling](https://github.com/sveltejs/kit/pull/9220/files#r1120243235) for the case where you're rendering the root page when `base` is set — if the base path is `/a/b/c` and you render `/a/b/c`, then the relative path (somewhat counter-intuitively) needs to be `./c` for links to work.

Turns out that was a mistake, because the root should always end with a trailing slash — if you prerender that page, it will be output as `<output>/index.html`, and relative paths will thus be incorrectly converted to `<output>/c/...`.

The fix is to remove the special case handling, and ensure that `/a/b/c` always redirects to `/a/b/c/`. This is much simpler and more correct. It arguably falls into the semver gray zone where someone somewhere might be relying on the buggy behaviour, but I think it's vanishingly unlikely, and in any case the worst that will happen is a URL that used to render something gets turned into a redirect.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

**Review by dummdidumm — COMMENTED** [review]
Would an alternative solution of checking whether or not we're on the base path with a trailing slash work, too? If you're on `/base/`, do like this fix does, if you're on `/base`, keep the old behavior.

**Comment by Rich-Harris** [comment]
> If you're on `/base/`, do like this fix does, if you're on `/base`, keep the old behavior.

`/base/` was already treated correctly, it was only `/base` where things (quite reasonably, in retrospect) got messed up

File: packages/kit/src/runtime/server/respond.js [hunk 1]
```diff
@@ -172,7 +172,9 @@ export async function respond(request, options, manifest, state) {
 	try {
 		// determine whether we need to redirect to add/remove a trailing slash
 		if (route && !is_data_request) {
-			if (route.page) {
+			if (url.pathname === base || url.pathname === base + '/') {
```
[the remark this answers is not in the document]
Comment (Rich-Harris) [reply packages/kit/src/runtime/server/respond.js:175]:
```suggestion
			// if `paths.base === '/a/b/c`, then the root route is `/a/b/c/`,
			// regardless of the `trailingSlash` route option
			if (url.pathname === base || url.pathname === base + '/') {
```

**Comment by mdi22** [comment]
Dear @Rich-Harris,

I tried the PR with my code, now.
Unfortunately, this PR only partly solves the problem (at least for me).

It works fine when I call the base path directly, in my case "http://localhost/build"

It also works following links to routes inside the already loaded SPA.

It does not work when calling a route below the base path in the browser directly, in my case "http://localhost/build/Termine" or "http://localhost/build/Termine/" (with or without trailing slash)

Am I missing something?

I'll attach the created scipt tag in index.html in different versions:
_URL "http://.../build/Termine"_

Sveltekit version **1.8**, working correctly:
```js
<script>
    {
        __sveltekit_rs4xzv = {
            env: {},
            assets: "/build",
            element: document.currentScript.parentElement
        };

        Promise.all([
            import("/build/_app/immutable/entry/start.3791b6e2.js"),
            import("/build/_app/immutable/entry/app.e98f8777.js")
        ]).then(([kit, app]) => {
            kit.start(app, __sveltekit_rs4xzv.element);
        });
    }
</script>
```

Sveltekit version **1.11**, neither working on base URL nor on path under base URL:
```js
<script>
    {
        __sveltekit_1h6aion = {
            env: {},
            assets: "/build",
            base: new URL("./build", location).pathname,
            element: document.currentScript.parentElement
        };

        Promise.all([
            import("./build/_app/immutable/entry/start.6980a169.js"),
            import("./build/_app/immutable/entry/app.35650362.js")
        ]).then(([kit, app]) => {
            kit.start(app, __sveltekit_1h6aion.element);
        });
    }
</script>
```

Sveltekit version PR 9343, working correctly on base URL, not on route under base URL:
```js
<script>
    {
        __sveltekit_inylau = {
            env: {},
            assets: "/build",
            base: new URL(".", location).pathname.slice(0, -1),
            element: document.currentScript.parentElement
        };

        Promise.all([
            import("./_app/immutable/entry/start.a989781b.js"),
            import("./_app/immutable/entry/app.3bd555c8.js")
        ]).then(([kit, app]) => {
            kit.start(app, __sveltekit_inylau.element);
        });
    }
</script>
```

My _paths / base_ setting is _'/build'_, but could be anything.

The problem seems to be that the path "./_app/immutable/entry/start.a989781b.js" is evaluated by the browser to
/build/Termine/_app/immutable/entry/start.a989781b.js

But only the file
/build/_app/immutable/entry/start.a989781b.js
was created

My .htaccess file is as follows
```
RewriteEngine On
RewriteCond %{REQUEST_FILENAME} !-f
RewriteRule .* index.html [NC,L]
```

I was not able to find a working RewriteRule to work around the new behavior.

Would it be the best to always use an absolute path when using paths/base?

Thank you already!

**Comment by Rich-Harris** [comment]
It sounds like you're rewriting all paths to `index.html`? If so, there's no reason to expect that to work! You need to use a `fallback` page (which must use absolute paths) if you're building an SPA.

**Pull request merged**